src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -418,6 +424,48 @@ public unsafe void SetResponseHeaders()
             }
         }
 
+        public unsafe void SetResponseTrailers()
+        {
+            HttpResponseTrailers.IsReadOnly = true;
+            foreach (var headerPair in HttpResponseTrailers)
+            {
+                var headerValues = headerPair.Value;
+
+                if (headerValues.Count == 0)
+                {
+                    continue;
+                }
+
+                string headerValue;
+                if (headerValues.Count == 1)
+                {
+                    headerValue = headerValues[0];
+                }
+                else
+                {
+                    var builder = new StringBuilder();
+                    builder.Append(headerValues[0]);
```

Comment:
I don't know how much you care about allocations here, but the new SB starts with a capacity of 16 and then starts doubling-ish. You could guess at what you need and presize it, or at least pass `headerValues[0]` to the constructor.

Reply:
Maybe 
```c#
var builder = new StringBuilder(headerValues[0], headerValues[0].Length * headerValues.Count)
```

Reply:
StringValues.ToString() already handles all of this for you.
https://github.com/dotnet/runtime/blob/7adab7c8114286539eed57781e7057eca13a566a/src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs#L167-L269

Reply:
And it's much more efficient 😄